### PR TITLE
[BEAM-7372] Drop Python 2 shims and update docstring in apache_beam.typehints.schemas

### PR DESCRIPTION
**Pull request by TheNeuralBit**
R: @ibzib 

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Dataflow | Flink | Samza | Spark | Twister2
--- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/) | ---
Java | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow_Java11/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink_Java11/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_SparkStructuredStreaming/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Twister2/lastCompletedBuild/)
Python | Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python36/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python37/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python38/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_VR_Dataflow_V2/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/) | ---
XLang | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Direct/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Flink/lastCompletedBuild/) | --- | [Build Status](https://ci-beam.apache.org/job/beam_PostCommit_XVR_Spark/lastCompletedBuild/) | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website | Whitespace | Typescript
--- | --- | --- | --- | --- | --- | ---
Non-portable | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonLint_Cron/lastCompletedBuild/)<br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocker_Cron/lastCompletedBuild/) <br>[Build Status](https://ci-beam.apache.org/job/beam_PreCommit_PythonDocs_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Whitespace_Cron/lastCompletedBuild/) | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Typescript_Cron/lastCompletedBuild/)
Portable | --- | [Build Status](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by TheNeuralBit** [comment]
Run Python PreCommit

**Comment by TheNeuralBit** [comment]
Failures look like known flakes in MacOS

File: sdks/python/apache_beam/typehints/schemas.py [hunk 1]
```diff
@@ -17,36 +17,42 @@
...
-ByteString  <-----> BYTES
+Imposes a mapping between common Python types and Beam portable schemas
+(https://s.apache.org/beam-schemas)::
+
+  Python              Schema
+  np.int8     <-----> BYTE
+  np.int16    <-----> INT16
+  np.int32    <-----> INT32
+  np.int64    <-----> INT64
+  int         ---/
```
Comment (ibzib) [inline sdks/python/apache_beam/typehints/schemas.py:28]: What does `---/ ` mean? That these types have no equivalent in schemas?

File: sdks/python/apache_beam/typehints/schemas.py [hunk 2]
```diff
@@ -17,36 +17,42 @@
...
+The mappings for :code:`STRING` and :code:`BYTES` are different between python
+2 and python 3, because of the changes to str.
+
+Python 3.x::
+
+  str/unicode <-----> STRING
+  bytes       <-----> BYTES
+  ByteString  ---/
+
+Python 2.x::
```
Comment (ibzib) [inline sdks/python/apache_beam/typehints/schemas.py:51]: Should we get rid of this since we're dropping Python 2 support?

**Comment by ibzib** [comment]
> Failures look like known flakes in MacOS

FYI I _think_ I fixed the flake in data_plane_test in #12637, so let me know if you see it again.

**Comment by TheNeuralBit** [comment]
> FYI I think I fixed the flake in data_plane_test in #12637, so let me know if you see it again.

Yep thank you for the fix! This run was before that was merged

File: sdks/python/apache_beam/typehints/schemas.py (hunk 1 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/typehints/schemas.py:28]: It's supposed to indicate that when converting to Beam schemas int will map to INT64, but it's not a two-way mapping. Converting back from Beam schemas will produce np.int64, not int.

Comment (ibzib) [reply sdks/python/apache_beam/typehints/schemas.py:28]: Wouldn't `------>` be a more obvious notation for that?

File: sdks/python/apache_beam/typehints/schemas.py (hunk 2 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/typehints/schemas.py:51]: Yeah I guess I should just drop all the Python 2 shims now  :tada:

File: sdks/python/apache_beam/typehints/schemas.py (hunk 1 above)
Comment (TheNeuralBit) [reply sdks/python/apache_beam/typehints/schemas.py:28]: I guess they seem pretty equivalent to me, but I see your point that it my ascii art for two arrows merging may not be obvious. I'll change it

File: sdks/python/apache_beam/typehints/schemas.py [hunk 3]
```diff
@@ -17,36 +17,36 @@
...
+  str/unicode <-----> STRING
+  bytes       <-----> BYTES
+  ByteString  ------> BYTES
+  Timestamp   <-----> LogicalType(urn="beam:logical_type:micros_instant:v1")
+  Mapping     <-----> MapType
+  Sequence    <-----> ArrayType
+  NamedTuple  <-----> RowType
+  beam.Row    ------> RowType
+
+Note that some of these mappings are provided as conveniences,
```
Comment (ibzib) [inline sdks/python/apache_beam/typehints/schemas.py:42]: Thanks for the note, this definitely helps.

Comment (TheNeuralBit) [reply sdks/python/apache_beam/typehints/schemas.py:42]: Thanks for helping me clarify my documentation :)

**Review by ibzib — APPROVED** [review]
LGTM

**Comment by TheNeuralBit** [comment]
Failure is due to flake BEAM-10866

**Pull request merged**